### PR TITLE
4759 organization layer administration ui base

### LINKED ISSUE #4759: Organizationlayer administration UI base
Organization layer administration base ui and components with moc data

Child of #4749

### PR DESCRIPTION
**Pull request by avukkonen**
Closes #4759

### CODE REVIEW

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/tabs.scss [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+  flex-basis: 50%;
+}
+
+.tabs__tab-data--workspace-students {
+
+  &.active {
+    display: flex;
+  }
+
+  @include breakpoint($breakpoint-pad) {
```
[the remark this answers is not in the document]
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/tabs.scss]: Whitespace removed. Overrides can't work like that, dear dog.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/tabs.scss [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+  &:first-child {
+    padding: 0 10px 0 0;
+  }
+
+  &:last-child {
+    padding: 0 0 0 10px;
+  }
+
+  @include breakpoint($breakpoint-pad) {
```
[the remark this answers is not in the document]
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/tabs.scss]: Nah

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/tabs.scss [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    border-top: solid 1px $color-application-panel-box-border;
+    padding: 10px 20px;
+  }
+  
+  &:first-child {
+    margin-left: -1px;
+    padding: 10px 20px;
+  }
+
+  @include breakpoint($breakpoint-pad) {
```
[the remark this answers is not in the document]
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/tabs.scss]: Nope

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/tabs.scss [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+  &:first-child {
+    padding: 0 10px 0 0;
+  }
+
+  &:last-child {
+    padding: 0 0 0 10px;
+  }
+
+  @include breakpoint($breakpoint-pad) {
```
[the remark this answers is not in the document]
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/tabs.scss]: no

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/tabs.scss [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+    border-top: solid 1px $color-application-panel-box-border;
+    padding: 10px 20px;
+  }
+  
+  &:first-child {
+    margin-left: -1px;
+    padding: 10px 20px;
+  }
+
+  @include breakpoint($breakpoint-pad) {
```
[the remark this answers is not in the document]
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/tabs.scss]: no

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/tabs.scss [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+    background: $color-tab-focus-background;
+    color: $color-tab-focus-font;
+    border-left: 1px solid $color-tab-border;
+    border-right: 1px solid $color-tab-border;
+    border-top: 1px solid $color-tab-border;
+    margin-bottom: -1px;
+    padding: 9px 10px 11px;
+  }
+
+  @include breakpoint($breakpoint-pad) {
```
[the remark this answers is not in the document]
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/tabs.scss]: njet

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/summary.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+
+class Summary extends React.Component<SummaryProps, SummaryState> {
+
+  
+  render(){
+    return (
+       <div>
+        <ApplicationSubPanel i18n={this.props.i18n} modifier="workspace-users" bodyModifier="workspace-staff-members" title={this.props.i18n.text.get('plugin.organization.summary.billing.title')}>
+          <ApplicationSubPanelItem i18n={this.props.i18n} title={this.props.i18n.text.get('plugin.organization.summary.billing.subtitle')} >Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec cursus enim libero, quis iaculis metus sollicitudin vel. In gravida facilisis felis, auctor commodo elit dignissim vel. Proin ac nisi faucibus, tincidunt turpis sed, interdum leo. Maecenas quis scelerisque dui. Sed scelerisque quam ut magna varius condimentum. Vivamus id venenatis neque. </ApplicationSubPanelItem>
```
Comment (onzag) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/summary.tsx]: What is this for exactly? Placeholder code?... Shouldn't it have been removed?...

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/courses.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+  
+}
+
+interface CoursesState {
+}
+
+const courses = [
+  {
+    name: "course",
+    description: "Morbi lacus tellus, scelerisque in lacus nec, ullamcorper suscipit velit. Curabitur arcu quam, laoreet sagittis posuere sit amet, dapibus a arcu. Integer tempor orci eu est feugiat, a mollis est aliquet. Donec fringilla sem a risus maximus volutpat",
```
Comment (onzag) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/courses.tsx]: Placeholder text again?... what is its purpose?...

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/courses.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+  },
+  ]
+
+class Courses extends React.Component<CoursesProps, CoursesState> {
+  
+  render(){
+    return (
+        <ApplicationList>
+          {courses.map((course:any, index)=>{
```
Comment (onzag) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/courses.tsx]: Please don't use the "any" key here, the usage of any should be avoided unless you are casting types or typescript becomes stubborn regarding defined data.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/courses/course.tsx [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+import '~/sass/elements/rich-text.scss';
+import '~/sass/elements/application-list.scss';
+import { StatusType } from '~/reducers/base/status';
+import {StateType} from '~/reducers';
+import { ApplicationListItem, ApplicationListItemHeader, ApplicationListItemBody, ApplicationListItemFooter } from '~/components/general/application-list';
+
+interface CourseProps {
+  i18n: i18nType,
+  status: StatusType,
+  course: any
```
Comment (onzag) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/courses/course.tsx]: Please define a type for the data you expect for these courses, if you expect name and description

interface ICourseType { name: string, description: string} and then course: would be ICourseType[]

Make sure there's no similar interface with the same name, just pick a name you think suits.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application.tsx [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+import Reports from './application/reports';
+import {i18nType} from '~/reducers/base/i18n';
+import { ButtonPill} from '~/components/general/button';
+import AddWorkspaceDialog from '../dialogs/new-workspace';
+import AddUserDialog from '../dialogs/new-user';
+
+
+import '~/sass/elements/link.scss';
+
+{/* Reading panel's css */}
```
Comment (onzag) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application.tsx]: Use a normal commenting style here

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application.tsx [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+  }
+  
+  onTabChange(id: "SUMMARY" | "USERS" | "COURSES" | "REPORTS" ) {
+    this.setState({
+      activeTab: id
+    });
+  }
+  
+  onInputFocus() {
+    console.log("Focus");
```
Comment (onzag) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application.tsx]: Debugging code?... what is this supposed to be used for?...

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/guider/body/application.tsx [hunk 13]
```diff
@@ -1,7 +1,7 @@
 import * as React from 'react';
 import {connect, Dispatch} from 'react-redux';
-
-import ApplicationPanel from '~/components/general/application-panel';
+//import ApplicationPanel from '~/components/general/application-panel';
```
Comment (onzag) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/guider/body/application.tsx]: If you moved it from one place to the other, and the previous general file is removed, then just remove the line

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/application-panel.tsx [hunk 14]
```diff
@@ -1,12 +1,19 @@
...
 
+
+// This is the deprecated application panel
+
 interface ApplicationPanelProps {
   modifier: string,
   title?: React.ReactElement<any> | string,
   icon?: React.ReactElement<any> | string,
+  panelTabs?: Array<TabType>
+//  tabs? : Array<TabType>,
```
Comment (onzag) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/application-panel.tsx]: If this property is gone just remove the line

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/discussion/body/application.tsx [hunk 15]
```diff
@@ -2,21 +2,18 @@ import * as React from 'react';
 import {connect, Dispatch} from 'react-redux';
 import {i18nType} from '~/reducers/base/i18n';
 import NewThread from '../dialogs/new-thread';
-import ApplicationPanel from '~/components/general/application-panel';
+//import ApplicationPanel from '~/components/general/application-panel';
```
Comment (onzag) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/discussion/body/application.tsx]: Again, if gone just remove the line

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/coursepicker/body/application.tsx [hunk 16]
```diff
@@ -1,14 +1,14 @@
 import * as React from 'react';
 import {connect, Dispatch} from 'react-redux';
-import ApplicationPanel from '~/components/general/application-panel';
+//import ApplicationPanel from '~/components/general/application-panel';
```
Comment (onzag) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/coursepicker/body/application.tsx]: Just remove the line

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/communicator/body/application.tsx [hunk 17]
```diff
@@ -1,20 +1,17 @@
 import * as React from 'react';
 import {connect, Dispatch} from 'react-redux';
-
-import ApplicationPanel from '~/components/general/application-panel';
+// import ApplicationPanel from '~/components/general/application-panel';
```
Comment (onzag) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/communicator/body/application.tsx]: Remove the line

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application.tsx [hunk 18]
```diff
@@ -1,7 +1,7 @@
 import * as React from 'react';
 import {connect, Dispatch} from 'react-redux';
-
-import ApplicationPanel from '~/components/general/application-panel';
+// import ApplicationPanel from '~/components/general/application-panel';
```
Comment (onzag) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/announcer/body/application.tsx]: Remove the line

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/courses.tsx (hunk 9 above)
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/courses.tsx]: I realized just now that I need to remove the mock data.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/summary.tsx (hunk 7 above)
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/summary.tsx]: Forgot to remove the mock up data.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/courses/course.tsx (hunk 10 above)
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/courses/course.tsx]: I do not have this

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application.tsx (hunk 11 above)
Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application.tsx]: I do not have this

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/users/usertabs.tsx [hunk 19]
```diff
@@ -0,0 +1,38 @@
...
+  constructor(props: OrganizationUserTabsProps){
+    super(props);
+    this.state = {
+      activeTab: "ACTIVE",
+    }
+    this.onTabChange = this.onTabChange.bind(this);
+  }
+  
+  onTabChange() {
+    alert("TAB!");
```
Comment (anttiviljakainen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/users/usertabs.tsx]: TAB!

Comment (avukkonen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/organization/body/application/users/usertabs.tsx]: The component wants a function to work, so I gave it this. I just tried to evade from removing things an putting them back after PR. But it's not a huge problem. Removed the component.

**Pull request merged**